webhooks/featureflagconfiguration_webhook.go

```diff
@@ -45,6 +47,16 @@ func (m *FeatureFlagConfigurationValidator) Handle(ctx context.Context, req admi
 			return admission.Denied(fmt.Sprintf("FeatureFlagSpec is not valid JSON: %s", err.Error()))
 		}
 	}
+
+	if config.Spec.Provider.Credentials.Name != "" {
```

Comment:
IMHO this should be part of the controller. I'm not sure what happens in a case where the provider and the flag configurations are applied at the same time and the configuration is faster ...

Reply:
Okay, we can move that logic there, I am not wed to this - I just thought putting the check in the webhook made sense

Reply:
@thschue I'm not sure I understand your concern :sweat_smile: 

> I'm not sure what happens in a case where the provider and the flag configurations are applied at the same time

If the CR instance doesn't yet exist, and it features the optional "provider" field, wouldn't we basically always be creating the provider and the flag spec at the "same time" since they are both parts of the same CR? And doesn't this webhook always run before any controller logic to create said resources, as @AlexsJones seems to be saying?

Can you clarify?

Reply:
I think @thschue was specifically referring to the modification of the object here
```
	if ffconf.Spec.Provider == nil {
		r.Log.Info("No provider specified for FeatureFlagConfiguration, using FlagD")
		ffconf.Spec.Provider = &corev1alpha1.FeatureFlagProvider{
			Name: "flagD",
		}
		if err := r.Update(ctx, ffconf); err != nil {
			r.Log.Error(err, "Failed to update FeatureFlagConfiguration Provider")
			return r.finishReconcile(err, false)
		}
	}

```

This however is in the controller and occurs after the webhook. 


The rationale for checking if the secret exists:

The secret will be mounted against the sidecar ( once we have a provider that requires it ) and if we don't check if it exists, it will break the deployments trying to mount a non-existent object.